packages/core/src/lib/theme.mts

```diff
@@ -3,18 +3,155 @@ import spinners from 'cli-spinners';
 import type { Prettify } from '@inquirer/type';
 
 type DefaultTheme = {
-  prefix: string;
+  /**
+   * Prefix to prepend to the message. If a function is provided, it will be
+   * called with the current status of the prompt (this will usually be
+   * `'pending'`, `'done'`, or `'loading'`), and the return value will be used
+   * as the prefix.
+   *
+   * @remarks
+   * If `status === 'loading'`, this property is ignored and the spinner (styled
+   * by the `spinner` property) will be displayed instead.
+   *
+   * @example
+   * ```ts
+   * (status) => status === 'done' ? colors.green('✔') : colors.green('?')
+   * ```
+   *
+   * @defaultValue
+   * ```ts
+   * // import colors from 'yoctocolors-cjs';
+   * colors.green('?')
+   * ```
+   */
+  prefix: string | ((status: string) => string);
+
+  /**
+   * Configuration for the spinner that is displayed when the prompt is in the
+   * `'loading'` state.
+   */
   spinner: {
+    /**
+     * The time interval between frames, in milliseconds.
+     *
+     * @defaultValue
+     * ```ts
+     * 80
+     * ```
+     */
     interval: number;
+
+    /**
+     * A list of frames to show for the spinner.
+     *
+     * @defaultValue
+     * ```ts
+     * ['⠋', '⠙', '⠹', '⠸', '⠼', '⠴', '⠦', '⠧', '⠇', '⠏']
+     * ```
+     */
     frames: string[];
   };
+  /**
+   * Object containing functions to style different parts of the prompt.
+   */
   style: {
+    /**
+     * Style to apply to the user's answer once it has been submitted.
+     *
+     * @param text - The user's answer.
+     * @returns The styled answer.
+     *
+     * @defaultValue
+     * ```ts
+     * // import colors from 'yoctocolors-cjs';
+     * (text) => colors.cyan(text)
+     * ```
+     */
     answer: (text: string) => string;
-    message: (text: string) => string;
+
+    /**
+     * Style to apply to the message displayed to the user.
+     *
+     * @param text - The message to style.
+     * @param status - The current status of the prompt.
+     * @returns The styled message.
+     *
+     * @defaultValue
+     * ```ts
+     * // import colors from 'yoctocolors-cjs';
+     * (text, status) => colors.bold(text)
+     * ```
+     */
+    message: (text: string, status: string) => string;
```

Comment:
Let's give `status` specific allowed strings. (and everywhere relevant, there's quite a few so I won't call them all 😅 )